src/ray/raylet/worker_pool.cc

```diff
@@ -890,6 +890,39 @@ std::shared_ptr<WorkerInterface> WorkerPool::PopWorker(
   return worker;
 }
 
+void WorkerPool::PrestartWorkers(const TaskSpecification &task_spec,
+                                 int64_t backlog_size) {
+  // Code path of task that needs a dedicated worker: an actor creation task with
+  // dynamic worker options, or any task with environment variable overrides.
+  if ((task_spec.IsActorCreationTask() && !task_spec.DynamicWorkerOptions().empty()) ||
+      task_spec.OverrideEnvironmentVariables().size() > 0) {
+    return;  // Not handled.
+  }
+
+  auto &state = GetStateForLanguage(task_spec.GetLanguage());
+  int num_idle_or_starting = state.idle.size();
+  for (auto &entry : state.starting_worker_processes) {
+    num_idle_or_starting += entry.second;
+  }
```

Comment:
I thought we should calculate the sum of all languages here at first because there are lots of counter variables here, and some cover only one language while others cover all languages. Could you add a comment here to explain the meaning of `num_idle_or_starting`?